pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1189,3 +1189,65 @@ func TestDeploymentTimedOut(t *testing.T) {
 		}
 	}
 }
+
+func TestMaxUnavailable(t *testing.T) {
```

Comment:
...and move these test to `TestResolveFenceposts` after moving the check to `ResolveFenceposts`